pkg/gatewayserver/gatewayserver.go

```diff
@@ -424,10 +424,16 @@ func (gs *GatewayServer) Connect(ctx context.Context, frontend io.Frontend, ids
 		return nil, err
 	}
 
+	addr := io.GatewayRemoteAddressFromContext(ctx)
```

Comment:
(I know an earlier comment was about this).
Nit: I think it would be better to get back to the variant that passed this via a parameter directly, and to just store it in the `io.Connection`.